sympy/physics/continuum_mechanics/beam.py

```diff
@@ -2297,298 +2296,660 @@ def plot_ild_moment(self,subs=None):
+            >>> b.apply_support(35, "roller")
+            >>> b.apply_support(70, "pin")
+            >>> b.draw()  # doctest: +SKIP
+        """
+
+        # Check if a load is negative or not
+        def _is_load_negative(load):
+            rv = load.is_negative
+            if load.is_Atom or rv is not None:
+                return rv
+            return load.doit().expand().is_negative
+
+        # Set up the plot with a decent size
+        fig, ax = plt.subplots(figsize=(10, 5.71))
+        ax.set_aspect("equal")
+
+        # Colors for different stuff
+        colors = {
+            "point_load": "blue",
+            "point_load_symbolic": "black",
+            "distributed_load": "#A30000",
+            "distributed_load_symbolic": "black",
+            "support_reaction": "#696969",
+        }
+
+        # Get the beam length, sub in 10 for any symbols
         if isinstance(self.length, Expr):
             l = list(self.length.atoms(Symbol))
-            # assigning every Symbol a default value of 10
             l = dict.fromkeys(l, 10)
-            length = self.length.subs(l)
+            length = float(self.length.subs(l))
         else:
             l = {}
-            length = self.length
-        height = length/10
-
-        rectangles = []
-        rectangles.append({'xy':(0, 0), 'width':length, 'height': height, 'facecolor':"brown"})
-        annotations, markers, load_eq,load_eq1, fill = self._draw_load(pictorial, length, l)
-        support_markers, support_rectangles = self._draw_supports(length, l)
-
-        rectangles += support_rectangles
-        markers += support_markers
-
-        for loc in self._applied_rotation_hinges:
-            ratio = loc / self.length
-            x_pos = float(ratio) * length
-            markers += [{'args':[[x_pos], [height / 2]], 'marker':'o', 'markersize':6, 'color':"white"}]
-
-        for loc in self._applied_sliding_hinges:
-            ratio = loc / self.length
-            x_pos = float(ratio) * length
-            markers += [{'args': [[x_pos], [height / 2]], 'marker':'|', 'markersize':12, 'color':"white"}]
-
-        ylim = (-length, 1.25*length)
-        if fill:
-            # when distributed loads are presents, they might get clipped out
-            # in the figure by the ylim settings.
-            # It might be necessary to compute new limits.
-            _min = min(min(fill["y2"]), min(r["xy"][1] for r in rectangles))
-            _max = max(max(fill["y1"]), max(r["xy"][1] for r in rectangles))
-            if (_min < ylim[0]) or (_max > ylim[1]):
-                offset = abs(_max - _min) * 0.1
-                ylim = (_min - offset, _max + offset)
-
-        sing_plot = plot(height + load_eq, height + load_eq1, (x, 0, length),
-            xlim=(-height, length + height), ylim=ylim,
-            annotations=annotations, markers=markers, rectangles=rectangles,
-            line_color='brown', fill=fill, axis=False, show=False)
```

Comment:
We want to use SymPy's plot function. If we don't "eat our own dog food" then the SymPy plotting will never be improved. If you need features that the SymPy plot module does not offer, then make PRs to SymPy plot() that adds them.